mycroft/tts/__init__.py

```diff
@@ -294,11 +294,15 @@ def __init__(self, tts):
         self.tts = tts
 
     def validate(self):
+        self.validate_dependencies()
         self.validate_instance()
         self.validate_filename()
         self.validate_lang()
         self.validate_connection()
 
+    def validate_dependencies(self):
```

Comment:
Is there a reason you implemented it this way instead of just making Boto3 a dependency in the "requirements.txt"?  I can think of a few reasons, just  curious about the why here.

Reply:
being a modular component i think it doesnt make sense to install all dependencies for all possible engines by default, we would end with many unused requirements

the user is already making a conscious choice when changing the engine, i think a warning for extra requirements works alright, maybe i could make it install the requirements automatically instead? 

i think this approach should be extended to TTS and Hotword (snowboy also isn't in  requirements.txt)

Reply:
I agree with you and think we have had a habit of semi-exposing too many options and sucking in a bunch of rarely used cruft.  For example, we have "ESpeak" as an option in Home although you can't use it unless you have setup a bunch of stuff manually on your system.

So I think I like the "validate_dependencies()" approach for those who are experimenting.  They will be able to see error messages in the logs this way.